pkg/security/tests/activity_tree_test.go

```diff
@@ -0,0 +1,3967 @@
+		pn.Process.Comm = path.Base(pn.Process.FileEvent.PathnameStr)
+		if pn.Process.Argv0 == "" {
+			pn.Process.Argv0 = pn.Process.Comm
+		}
+		pn.Process.ContainerID = "123"
+	}
+	return pns
+}
+
+// newExecTestEventWithAncestors returns a new exec test event with a process cache entry populated with the input list.
+// A final `systemd` node is appended.
+func newExecTestEventWithAncestors(lineage []model.Process) *model.Event {
+	// add missing fields
+	for i := range lineage {
+		lineage[i].FileEvent.BasenameStr = path.Base(lineage[i].FileEvent.PathnameStr)
+		lineage[i].Comm = lineage[i].FileEvent.BasenameStr
+		if lineage[i].Argv0 == "" {
+			lineage[i].Argv0 = lineage[i].Comm
+		}
+		lineage[i].Pid = uint32(lineage[i].FileEvent.Inode + 10)
+	}
+
+	// build the list of ancestors
+	ancestor := new(model.ProcessCacheEntry)
+	lineageDup := make([]model.Process, len(lineage))
+	copy(lineageDup, lineage)
+
+	// reverse lineageDup
+	for i, j := 0, len(lineageDup)-1; i < j; i, j = i+1, j-1 {
+		lineageDup[i], lineageDup[j] = lineageDup[j], lineageDup[i]
+	}
+
+	cursor := ancestor
+	for _, p := range lineageDup[1:] {
+		cursor.Process = p
+		cursor.Ancestor = new(model.ProcessCacheEntry)
+		cursor = cursor.Ancestor
+	}
+
+	// append systemd
+	cursor.Process = model.Process{
+		PIDContext: model.PIDContext{
+			Pid: 1,
+		},
+		Comm: "systemd",
+		FileEvent: model.FileEvent{
+			PathnameStr: "/bin/systemd",
+			BasenameStr: "systemd",
+			FileFields: model.FileFields{
+				PathKey: model.PathKey{
+					Inode: 42,
+				},
+			},
+		},
+	}
+
+	evt := &model.Event{
+		BaseEvent: model.BaseEvent{
+			Type:             uint32(model.ExecEventType),
+			FieldHandlers:    &model.DefaultFieldHandlers{},
+			ContainerContext: &model.ContainerContext{},
+			ProcessContext:   &model.ProcessContext{},
+			ProcessCacheEntry: &model.ProcessCacheEntry{
+				ProcessContext: model.ProcessContext{
+					Process:  lineageDup[0],
+					Ancestor: ancestor,
+				},
+			},
+		},
+		Exec: model.ExecEvent{
+			Process: &model.Process{},
+		},
+	}
+	evt.ProcessContext = &evt.ProcessCacheEntry.ProcessContext
+	evt.Exec.Process = &evt.ProcessContext.Process
+	return evt
+}
+
+// InsertExecEventTestCase defines an activity-tree insert exec-event test case
+type InsertExecEventTestCase struct {
+	Name         string
+	Tree         *activity_tree.ActivityTree
+	InputEvent   *model.Event
+	wantNewEntry bool
+	WantTree     *activity_tree.ActivityTree
+	wantNode     *activity_tree.ProcessNode
+}
+
+// InsertExecEventTestCases defines all the activity-tree insert exec-event test cases
+var InsertExecEventTestCases = []InsertExecEventTestCase{
+	// exec/01
+	// ---------------
+	//
+	//     empty tree          +          systemd                 ==>>              /bin/bash
+	//                                       |- /bin/bash                               |
+	//                                       |- /bin/ls                              /bin/ls
+	{
+		Name: "exec/01",
```

Comment:
I think it would be better to keep only the json version.

Reply:
We still don't have stable merge/insertion algo, the logic may change again, and IMHO it would be way easier to add/modify existing tests using structs instead of editing manually json files. But I agree that at some point we should get rid of them.

Hmm, I think if we really want to remove them right now, we could also keep a tiny side tool to generate json files.